Ex1.java

```diff
@@ -0,0 +1,134 @@
+import java.util.HashMap;
+
+public class Ex1 {
+
+    /**
+     * The function receives two strings and checks if one is a permutation
+     * of the other
+     * @param firstStr
+     * @param secondStr
+     * @return true if one is a permutation of the other, otherwise return false
+     */
+    public static boolean isPermutation (String firstStr, String secondStr){
+        // Check if one of the strings are null or if their length is not equal
+        if (firstStr == null || secondStr == null || firstStr.length() != secondStr.length()){
+            return false;
+        }
+
+        HashMap<Character, Integer> map = new HashMap<Character, Integer>();
+
+        // Create a hash map so that the key is the character
+        // and the value is the number of times that the character appears in the string
+        for(int i = 0; i < firstStr.length(); i++){
+            Character currChar = Character.toLowerCase(firstStr.charAt(i));
+            // Check if the map contains a mapping for the specified character
+            if (map.containsKey(currChar)){

```

Comment:
Optional way to simplify the whole block: [Map.getOrDefault](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#getOrDefault-java.lang.Object-V-). Could also be applied to the second if below.

Feel free to leave it as is though if you prefer explicit if.